src/shared/constants.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Utility methods to perform common ops with constant values like:
+ *  - Birth year
+ *  - Gender
+ *  - Disability
+ *  - Etc
+ */
+import moment from 'moment';
+
+
+const capitaliseFirstWord = s => s.replace(/^\w/, str => str.toUpperCase());
+
+export const BirthYear = {
+  // list :: (Number, Number, ['desc' | 'asc']) -> [Number]
+  list: (start, end, order = 'desc') =>
+    Array.from({ length: (end + 1) - start }, (v, i) => (order === 'desc') ? end - i : start + i),
+
+  // listToOptions :: (Number) -> [{ key: string, value: string }]
+  listToOptions: years =>
+    [{ key: '', value: '' }]
+      .concat(
+        years.map(y => ({ key: String(y), value: String(y) })),
+      ),
+
+  // defaultOptionsList :: () -> [{ key: string, value: string }]
+  defaultOptionsList: () =>
+    BirthYear.listToOptions(
+      BirthYear.list(new Date().getFullYear() - 113, new Date().getFullYear()),
+    ),
+
+  // toAge :: Number | null -> Number | null
+  toAge: year =>
+    year === null ? null : moment().year() - year,
+
+  // fromAge :: Number | null -> Number | null
+  fromAge: age =>
+    age === null ? null : moment().year() - age,
+
+  // toDisplay :: Number | null -> String
+  toDisplay: year =>
+    year === null ? 'Prefer not to say' : year,
```

Comment:
I think it should be the responsibility of the server to do this, not the client.

Reply:
IMO the proper place for presentational logic is the client/frontend code.

Reply:
My preference for this being handled by the server is due to how filtering and aggregating data will look.

- requests can be filtered by age. If this is handled by the client then queries will be age=null
- aggregated data is handled by the server and will return data for age group 'prefer not to say'.

Reply:
> aggregated data is handled by the server and will return data for age group 'prefer not to say'.

I may be missing a piece here, so forgive me.

Are we talking exclusively about `/community-bussinesses/me/visit-logs/aggregates` and `/community-businesses/me/visitors`?

The age filter is currently implemented as a two-element array of numbers. This will need to change in any case, since it doesn't support the new use case.

I don't see why we would prefer `?age=prefer not to say` instead of `?age=null`. I know I said differently [here](https://github.com/TwinePlatform/twine-visitor/issues/584#issuecomment-448624123), but that was just my first reaction. I still think ->

> IMO the proper place for presentational logic is the client/frontend code.